index_batcher.go

```diff
@@ -0,0 +1,164 @@
+package bleve
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+// NOTE XXX As this approach uses carefully orchestrated interactions
+// between a mutex, channels, and reference swapping, be sure to carefully
+// consider the impact of reordering statements or making changes to the
+// patterns present in the code.
+
+// bleveIndex is an alias that allows the IndexBatcher to embed a Index
+type bleveIndex Index
+
+// IndexBatcher can be wrapped around a Index to aggregate operations
+// in a concurrent / parallel context for increased throughput.
+type IndexBatcher struct {
+	bleveIndex
+
+	period time.Duration
+	closer chan bool
+
+	// lock is used to protect applying / resetting the batch, updating /
+	// replacing the result, and signalling / replacing the signal channel from
+	// the batch loop.  Elsewhere it is used for getting a reference to the
+	// current result, getting a reference to the current signal channel,
+	// and adding operations to the batch.
+	lock   sync.Mutex
+	batch  *Batch
+	result *rslt
+	signal chan bool
+}
+
+// rslt is used as a kind of double pointer for errors
```

Comment:
I still don't quite get the need for this.  Locally I just changed it to `error` and it compiles and the test passes.  Can you explain?